.gitlab-ci.yml

```
@@ -31,21 +50,15 @@ integration-testing:
     KUBERNETES_MEMORY_REQUEST: "6Gi"
     KUBERNETES_MEMORY_LIMIT: "12Gi"
 
-build-runner-image:
-  stage: build
+release-runner-image:
+  stage: release
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/docker:20.10.13
   tags: ["arch:amd64"]
-  variables:
-    PUSH_IMAGE: "false"
-  rules:
-    - if: $CI_COMMIT_BRANCH == "main"
-      variables:
-        PUSH_IMAGE: "true"
-    - when: on_success
   script:
-    - TARGET_REPO="${IMAGE_REPOSITORY}/runner"
-    - if [ "$PUSH_IMAGE" = "true" ]; then PUSH_ARG="--push"; fi
-    # Tag with 8 and 12 SHA to match go.mod
-    - docker buildx build --no-cache --pull $PUSH_ARG --tag ${TARGET_REPO}:${CI_COMMIT_SHORT_SHA} --tag ${TARGET_REPO}:${CI_COMMIT_SHA:0:12}  .
-  retry: 2
-
+    - docker pull ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA}
+    - docker tag  ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA} ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHORT_SHA} 
+    - docker tag  ${CI_REGISTRY_IMAGE_TEST}:${CI_COMMIT_SHORT_SHA} ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA:0:12}
+    - docker push --label target=build ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA:0:12} --tag ${IMAGE_REPOSITORY}:${CI_COMMIT_SHA:0:12}  .
```

```suggestion
    - docker push ${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHA:0:12}
```

The `docker push` syntax does not work as `docker build`, it only takes a single parameter and does not take any working dir path.

Also I believe we can now remove the `${CI_REGISTRY_IMAGE}:${CI_COMMIT_SHORT_SHA}` tag as it should not be used anywhere anymore.

---

Forgot to update it, the label is added at the previous build stage, tags are added with `docker tag`